src/plugins/explore/public/application/pages/traces/trace_details/public/gantt_chart_vega/gantt_chart_vega.tsx

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import { View, parse } from 'vega';
+import { createGanttSpec } from './gantt_chart_spec';
+import { convertToVegaGanttData } from './gantt_data_adapter';
+import { GANTT_CHART_CONSTANTS, TOTAL_PADDING } from './gantt_constants';
+import './gantt_chart_vega.scss';
+
+interface GanttChartProps {
+  data: any[];
+  colorMap: Record<string, string>;
+  height: number;
+  onSpanClick?: (spanId: string) => void;
+  selectedSpanId?: string;
+}
+
+export function GanttChart({
+  data,
+  colorMap,
+  height,
+  onSpanClick,
+  selectedSpanId,
+}: GanttChartProps) {
+  const containerRef = useRef<HTMLDivElement>(null);
+  const viewRef = useRef<View | null>(null);
+  const [containerWidth, setContainerWidth] = useState<number>(800);
```

Comment:
you know better than me, but is this state unavoidable? vega can't automatically deduce the width of the container?

Reply:
Vega's built in spacing does not interface well with the gantt chart bars being dependent on width.
Will look into possible refactor as follow-up but haven't been able to get the built in resizing to work properly.